.github/workflows/pre-release.yml

```
@@ -0,0 +1,168 @@
+name: Pre-Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      new_version:
+        description: 'The new release version (e.g., v2.10.0)'
+        required: true
+        type: string
+
+permissions:
+  contents: write
+  pull-requests: write
+  
+env:
+  GO_VERSION: "^1.24.6"
+  GOLANGCI_LINT_VERSION: "v2.4.0"
+  E2E_SETUP_KIND: yes
+  E2E_SETUP_KUBECTL: yes
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Validate version format'
+        run: |
+          if [[ ! "${{ inputs.new_version }}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "Error: Version format is incorrect. It must be in the format 'vX.Y.Z'."
+            exit 1
+          fi
+          echo "Version format is valid."
+      - name: Checkout into the corresponding release branch
+        uses: actions/checkout@v4
+      - name: Create VCS sandbox
+        run: |
+          git checkout -b release-prep-${{ inputs.new_version }}
+          MAJOR_MINOR_VERSION=$(echo "${{ inputs.new_version }}" | sed 's/^v//' | cut -d. -f1,2)
+          if git show-ref --verify --quiet refs/remotes/origin/release-$MAJOR_MINOR_VERSION; then
+            echo "Release branch release-$MAJOR_MINOR_VERSION already exists, switching to it"
+            git checkout -b release-$MAJOR_MINOR_VERSION origin/release-$MAJOR_MINOR_VERSION
+          else
+            echo "Creating new release branch release-$MAJOR_MINOR_VERSION"
+            git checkout -b release-$MAJOR_MINOR_VERSION
+            git push origin release-$MAJOR_MINOR_VERSION
+          fi          
+          git checkout release-prep-${{ inputs.new_version }}
+      - name: Set up the Go@${{ env.GO_VERSION }} environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Update the VERSION manifest
+        run: echo "${{ inputs.new_version }}" | sed 's/^v//' > VERSION
+      - name: update data.yaml
+        run: |
+            chmod +x scripts/update-data-yaml.sh
+            ./scripts/update-data-yaml.sh "${{ inputs.new_version }}"
+      - name: Update the compatibility matrix (README.md)
+        run: make generate
+      - name: Generate the release date for unreleased (CHANGELOG.md)
+        run: |
+          chmod +x scripts/generate-release-date.sh
+          ./scripts/generate-release-date.sh "${{ inputs.new_version }}"
+      - name: Lint
```

```suggestion
      - name: make lint-fix
```

Ideally the name should reflect the difference between lint and lint-fix targets.